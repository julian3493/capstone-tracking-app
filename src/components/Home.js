import React from 'react';
import Registration from './Registration';

const Home = () => (
  <div>
    <h1>Home</h1>
    <Registration />
  </div>
);

export default Home;
