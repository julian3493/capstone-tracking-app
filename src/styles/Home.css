.options {
  margin: 50px 30px;
  padding: 40px 15px;
  background-color: #f3f4f6;
  display: flex;
  flex-direction: column;
  align-items: center;
}

button {
  background-color: #41b5e8;
  color: white;
  width: 50%;
  border-radius: 5%;
  margin-bottom: 10px;
  padding: 2%;
}

a {
  color: white;
  font-size: 1rem;
}
