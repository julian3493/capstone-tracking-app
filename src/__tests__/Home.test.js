import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Home from '../components/Home';
import store from '../redux/store';

it('it tests the render Home correctly', () => {
  const home = renderer
    .create(<Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>)
    .toJSON();
  expect(home).toMatchSnapshot();
});


describe('render correctly the topbar component', () => {
  it('renders the topbar', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const topbar = document.querySelector('.top');
    expect(topbar).toBeDefined();
  });

  it('renders the page title', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const topbar = document.querySelector('.top');
    expect(topbar.firstChild.innerHTML).toEqual('App Tracker');
  });
});

describe('render correctly the login and create buttons', () => {
  it('show a login button', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const options = document.querySelector('.options');
    expect(options.firstChild.innerHTML).toEqual('<a href=\"/login\">Login</a>');
  });

  it('show a create account button', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const options = document.querySelector('.options');
    expect(options.lastChild.innerHTML).toEqual('<a href=\"/registration\">Create Account</a>');
  });
});