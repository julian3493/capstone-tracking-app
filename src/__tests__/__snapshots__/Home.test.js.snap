// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the render Home correctly 1`] = `
<div>
  <div
    className="top"
  >
    <p
      className="top-title"
    >
      App Tracker
    </p>
  </div>
  <div
    className="options"
  >
    <button
      type="button"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </button>
    <button
      type="button"
    >
      <a
        href="/registration"
        onClick={[Function]}
      >
        Create Account
      </a>
    </button>
  </div>
</div>
`;
